src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useMemo } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem';
+
+type Props = {
+  visibleToDos: Todo[];
+  onDelete: (id: number) => Promise<void>;
+  onUpdate: (id: number, updated: Partial<Todo>) => void;
+  tempTodo: Todo | null;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  onDelete,
+  onUpdate,
+  tempTodo,
+}) => {
+  const renderToDos = useMemo(
+    () => (tempTodo ? [...visibleToDos, tempTodo] : visibleToDos),
+    [visibleToDos, tempTodo],
+  );
```

Comment:
fix linter errors and warnings 
![image](https://github.com/user-attachments/assets/3f4fd505-e0d4-4a60-a472-6544f987a98d)